sources/niLang/src/Clipboard.cpp

```diff
@@ -78,11 +67,19 @@ tBool __stdcall cLang::SetClipboard(eClipboardType aType, iDataTable* apDT) {
   return eTrue;
 }
 
-///////////////////////////////////////////////
-iDataTable* __stdcall cLang::GetClipboard(eClipboardType aType) const {
+// GetClipboard can be async so we recieve a callback.
+iDataTable* __stdcall cLang::GetClipboard(eClipboardType aType, iCallback* callback) const {
   niCheck(aType < eClipboardType_Last,NULL);
   if (aType == eClipboardType_System) {
     niThis(cLang)->mptrClipboard[aType] = _GetSystemClipboard(mptrClipboard[aType]);
   }
+  if (callback) {
+    callback->RunCallback(mptrClipboard[aType], niVarNull);
```

Comment:
Pass "aType" as 2nd parameter. Cast it to (tU32) if the compiler complains about the enum.